zebra-state/src/service/check/anchors.rs

```diff
@@ -214,119 +411,68 @@ pub(crate) fn sprout_anchors_refer_to_treestates(
-            // > note commitments and nullifiers specified in that JoinSplit
-            // > description to the input treestate referred to by its
-            // > anchor. This interstitial output treestate is available for
-            // > use as the anchor of subsequent JoinSplit descriptions in
-            // > the same transaction.
-            //
-            // <https://zips.z.cash/protocol/protocol.pdf#joinsplit>
-            //
-            // # Consensus
-            //
-            // > The anchor of each JoinSplit description in a transaction
-            // > MUST refer to either some earlier block’s final Sprout
-            // > treestate, or to the interstitial output treestate of any
-            // > prior JoinSplit description in the same transaction.
-            //
-            // > For the first JoinSplit description of a transaction, the
-            // > anchor MUST be the output Sprout treestate of a previous
-            // > block.
-            //
-            // <https://zips.z.cash/protocol/protocol.pdf#joinsplit>
-            //
-            // Note that in order to satisfy the latter consensus rule above,
-            // [`interstitial_trees`] is always empty in the first iteration
-            // of the loop.
-            let input_tree = interstitial_trees
-                .get(&joinsplit.anchor)
-                .cloned()
-                .or_else(|| sprout_final_treestates.get(&joinsplit.anchor).cloned());
-
-            tracing::trace!(
-                ?input_tree,
-                final_lookup = ?sprout_final_treestates.get(&joinsplit.anchor),
-                interstitial_lookup = ?interstitial_trees.get(&joinsplit.anchor),
-                interstitial_tree_count = ?interstitial_trees.len(),
-                ?interstitial_trees,
-                ?height,
-                "looked up sprout treestate anchor",
-            );
-
-            let mut input_tree = match input_tree {
-                Some(tree) => tree,
-                None => {
-                    tracing::debug!(
-                        ?joinsplit.anchor,
-                        ?joinsplit_index_in_tx,
-                        ?tx_index_in_block,
-                        ?height,
-                        "failed to find sprout anchor",
-                    );
-                    return Err(ValidateContextError::UnknownSproutAnchor {
-                        anchor: joinsplit.anchor,
-                        height,
-                        tx_index_in_block,
-                        transaction_hash: transaction_hashes[tx_index_in_block],
-                    });
-                }
-            };
-
-            tracing::debug!(
-                ?joinsplit.anchor,
-                ?joinsplit_index_in_tx,
-                ?tx_index_in_block,
-                ?height,
-                "validated sprout anchor",
-            );
+    block
+        .transactions
+        .par_iter()
```

Comment:
I think if `sprout_final_treestates.is_empty()` it would return `UnknownSproutAnchor` in the first iteration of the loop in `sprout_anchors_refer_to_treestates`